docs/sql-ref-functions-builtin-aggregate.md

```diff
@@ -19,4 +19,628 @@ license: |
+      <td> Returns the average of values in the input column.</td> 
+    </tr>
+    <tr>
+      <td> <b>{bool_and | every}</b>(<i>c: Column</i>)</td>
+      <td>Column name</td>
+      <td>Returns true if all values are true</td>
+    </tr>
+    <tr>
+      <td> <b>collect_list</b>(<i>c: Column</i>)</td>
+      <td>Column name</td>
+      <td>Collects and returns a list of non-unique elements. The function is non-deterministic because the order of collected results depends on the order of the rows which may be non-deterministic after a shuffle</td>
+    </tr>       
+    <tr>
+      <td> <b>collect_set</b>(<i>c: Column</i>)</td>
+      <td>Column name</td>
+      <td>Collects and returns a set of unique elements. The function is non-deterministic because the order of collected results depends on the order of the rows which may be non-deterministic after a shuffle.</td>
+    </tr>
+    <tr>
+      <td> <b>corr</b>(<i>c1: Column, c2: Column</i>)</td>
+      <td>Column name</td>
+      <td>Returns Pearson coefficient of correlation between a set of number pairs</td>
+    </tr>
+    <tr>
+      <td> <b>count</b>(<i>*</i>)</td>
+      <td>None</td>
+      <td>Returns the total number of retrieved rows, including rows containing null</td>
+    </tr>
+    <tr>
+      <td> <b>count</b>(<i>c: Column[, c: Column]</i>)</td>
+      <td>Column name</td>
+      <td>Returns the number of rows for which the supplied column(s) are all not null</td>
+    </tr>
+    <tr>
+      <td> <b>count</b>(<b>DISTINCT</b> <i> c: Column[, c: Column</i>])</td>
+      <td>Column name</td>
+      <td>Returns the number of rows for which the supplied column(s) are unique and not null</td>
+    </tr> 
+    <tr>
+      <td> <b>count_if</b>(<i>Predicate</i>)</td>
+      <td>Expression that will be used for aggregation calculation</td>
+      <td>Returns the count number from the predicate evaluate to <code>TRUE</code> values</td>
+    </tr> 
+    <tr>
+        <td> <b>count_min_sketch</b>(<i>c: Column, eps: double, confidence: double, seed integer</i>)</td>
+        <td>Column name; eps is a value between 0.0 and 1.0; confidence is a value between 0.0 and 1.0; seed is a positive integer</td>
+        <td>Returns a count-min sketch of a column with the given esp, confidence and seed. The result is an array of bytes, which can be deserialized to a `CountMinSketch` before usage. Count-min sketch is a probabilistic data structure used for cardinality estimation using sub-linear space..</td>
+    </tr>
+    <tr>
+      <td> <b>covar_pop</b>(<i>c1: Column, c2: Column</i>)</td>
+      <td>Column name</td>
+      <td>Returns the population covariance of a set of number pairs</td>
+    </tr> 
+    <tr>
+      <td> <b>covar_samp</b>(<i>c1: Column, c2: Column</i>)</td>
+      <td>Column name</td>
+      <td>Returns the sample covariance of a set of number pairs</td>
+    </tr>  
+    <tr>
+      <td> <b>{first | first_value}</b>(<i>c: Column[, isIgnoreNull]</i>)</td>
```

Comment:
`isIgnoreNull` -> `isIgnoreNull: Boolean`? btw, what does `Column` mean? I think we need to use concrete SQL types. How about following the PostgreSQL docs? https://www.postgresql.org/docs/current/functions-aggregate.html

Reply:
I was thinking that `Column` is a type in spark, it is not a concrete type. I changed to use `expression`, maybe it is a better name. I changed the table format to use the concrete SQL types. I used the concrete type which internal code is checking, although they may take other data type as input(spark will try to cast).

Reply:
Ah, the current one looks better! Thanks for the work!

Reply:
btw, I think its better to use the same type names here with https://github.com/apache/spark/blob/master/docs/sql-ref-datatypes.md

Reply:
Does my current type name look ok?  `boolean` , `numeric` , `string`